### PR TITLE
[Merged by Bors] - feat: port Algebra.Category.Algebra.Basic

### PR DESCRIPTION
**Pull request by 4e554c4c**
---

- [x] depends on: #4693  [concrete category fixes]

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by 4e554c4c** [comment]
i believe we should wait for #4693 to merge before this file is ported

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4693
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4693~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by jcommelin** [comment]
Files builds, but there are some linter failures.

File: Mathlib/Algebra/Category/AlgebraCat/Basic.lean [hunk 1]
```diff
@@ -202,9 +214,9 @@ def toAlgEquiv {X Y : AlgebraCat R} (i : X ≅ Y) : X ≃ₐ[R] Y where
     change (i.inv ≫ i.hom) x = x
     simp only [inv_hom_id]
     rw [id_apply]
-  map_add' := by simp
-  map_mul' := by simp
-  commutes' := by simp
+  map_add' := i.hom.map_add -- Porting note: was `by simp`
+  map_mul' := i.hom.map_mul -- Porting note: was `by simp`
+  commutes' := i.hom.commutes -- Porting note: was `by simp`
```
Comment (4e554c4c) [inline Mathlib/Algebra/Category/AlgebraCat/Basic.lean:219]: these `by simp` at the end used to be `by tidy`, I was playing around with alternatives and didn't leave a porting note, so we should probably change the porting note you added.

Comment (jcommelin) [reply Mathlib/Algebra/Category/AlgebraCat/Basic.lean]: Aah, good catch. Please go ahead!

File: Mathlib/Algebra/Category/AlgebraCat/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+@[simp]
+theorem coe_comp (f : M ⟶ N) (g : N ⟶ U) : (f ≫ g : M → U) = g ∘ f :=
+  rfl
+#align Algebra.coe_comp AlgebraCat.coe_comp
+
+variable (R)
+
+@[simp]
+theorem simps.free_obj_isRing_nsmul (a : ℕ) (b : FreeAlgebra R S) :
```
Comment (4e554c4c) [inline Mathlib/Algebra/Category/AlgebraCat/Basic.lean:136]: should we get rid of these new simps lemmas since they aren't needed anymore?

Comment (jcommelin) [reply Mathlib/Algebra/Category/AlgebraCat/Basic.lean:136]: Yes please

File: Mathlib/Algebra/Category/AlgebraCat/Basic.lean [hunk 3]
```diff
@@ -117,19 +132,36 @@ theorem coe_comp (f : M ⟶ N) (g : N ⟶ U) : (f ≫ g : M → U) = g ∘ f :=
...
+ ↑a * b = nsmulRec a b := by
+  -- simp only [@nsmul_eq_mul, AlgebraCat.free_obj_isRing_natCast, AlgebraCat.free_obj_isRing_mul]
+  rw [← nsmul_eq_mul]; rfl
+
+@[simp]
+theorem simps.free_obj_isAlgebra_toFun (r : R) :
+  (Algebra.toRingHom r : FreeAlgebra R S) =
+    Quot.mk (FreeAlgebra.Rel R S) (FreeAlgebra.Pre.of_scalar r) :=
+by rfl
+
```
Comment (jcommelin) [inline Mathlib/Algebra/Category/AlgebraCat/Basic.lean:146]:
```suggestion
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**